src/lib.rs

```diff
@@ -297,30 +292,50 @@ pub fn run(config: Config) -> Result<(), Box<dyn Error>> {
                 &config.pane_name_user_option,
                 &pane.name.clone(),
             );
-            let error_message = "Unable to run rename pane command.";
             if let Some(rename_pane_args_) = rename_pane_args {
-                run_tmux_command(&rename_pane_args_, error_message);
+                commands.push(rename_pane_args_.clone());
             }
         }
 
         let window_layout_args =
             build_window_layout_args(session_name, &window_index, &config.layout, &window.layout);
 
         if let Some(window_layout_args_) = window_layout_args {
-            let error_message = "Unable to set window layout.";
-            run_tmux_command(&window_layout_args_, error_message)
+            commands.push(window_layout_args_.clone());
         }
     }
 
+    commands
+}
+
+pub fn run_start(config: Config) -> Result<(), Box<dyn Error>> {
+    let commands = convert_config_to_tmux_commands(&config);
+
+    for command in commands.iter() {
+        run_tmux_command(&command);
+    }
+
     // TODO: Move this into helper. First attempt resulted in error caused by
     // return type. I think I either need to return the command and then spawn
     // or return the result of calling spawn.
-    let attach_args = build_attach_args(&session_name);
+    let attach_args = build_attach_args(&config.name);
     let _attach_output = Command::new("tmux").args(&attach_args).spawn()?.wait();
 
     Ok(())
 }
 
+pub fn run_debug(config: Config) -> Result<(), Box<dyn Error>> {
+    let commands = convert_config_to_tmux_commands(&config)
+        .iter()
+        .map(|v| v.join(" "))
+        .collect::<Vec<String>>();
+    for command in commands.iter() {
+        println!("tmux {}", command);
+    }
```

Comment:
Looks like I forgot to submit this comment the first time, so here we go again.

So just as a heads up you can actually just use `Vec<_>` in the turbofish instead of the full type. `.collect()` is very generic over its return type (anything that implements `FromIterator`) so you really just need to specify the container for it to use instead of the full type signature.

And even though I really love using iterator changes in Rust I think this would be simpler if you just used a `for-loop`. You have the right idea of not trying to `println` in the iterator which is good because Rust's iterators are lazy so it wouldn't be guaranteed to actually run, but the above could be represented as just

```rust
for command in convert_config_to_tmux_commands(&config) {
    println!("tmux {}", command.join(" "));
}
```

Since using the iterator and the for loop is kinda pulling double duty.